packages/contracts/contracts/protocol/oracles/BalancerOracle.sol

```diff
@@ -121,7 +96,22 @@ library BalancerOracle {
 			if (i == bptIndex) {
 				continue;
 			}
-			uint256 rateAdjustedPrice = getRateAdjustedPrice(vmexOracle, address(tokens[i]), rateProviders[i],pool, legacy);
+			// Get the price of each of the base tokens in ETH
+			// This also includes the price of the nested LP tokens, if they are e.g. LinearPools
+			// The only requirement is that the nested LP tokens have a price oracle registered
+			// See BalancerLpLinearPoolPriceOracle.sol for an example, as well as the relevant tests
+			uint256 price = IPriceOracle(vmexOracle).getAssetPrice(address(tokens[i]));
+			uint256 tokenRate;
+			if(legacy) {
+				if(rateProviders[i] == address(0)){
+					return price;
+				} else {
+					tokenRate = IRateProvider(rateProviders[i]).getRate();
+				}
+			} else {
+				tokenRate = pool.getTokenRate(address(tokens[i]));
+			}
+			uint256 rateAdjustedPrice = (price * 1e18) / tokenRate; //rate always has 18 decimals, so this preserves original decimals of price
 			if (rateAdjustedPrice < minPrice) {
```

Comment:
I dont understand this:

`		
uint256 minPrice = type(uint256).max;

			uint256 rateAdjustedPrice = (price * 1e18) / tokenRate; //rate always has 18 decimals, so this preserves original decimals of price
			
			if (rateAdjustedPrice < minPrice) {
				minPrice = rateAdjustedPrice;
			}
			
					return (minPrice * pool.getRate()) / 1e18;
`


In what case will rateAdjustedPrice be max uint256? from what I see its divided by tokenRate, so its never going to be max uint256. and if it is, then minPrice * pool.getRate() will revert, because max uint256 x any number > 1 is overflow

Reply:
minPrice is set as the max first so it can find the min. It will never be max cause any rateAdjustedPrice will be less than the max so it will override the minPrice